engines/errors.go

```diff
@@ -26,6 +26,9 @@ var ErrImmutableMountNotSupported = errors.New("The engine doesn't support immut
 // doesn't exist.
 var ErrResourceNotFound = errors.New("The referenced resource wasn't found")
 
+// ErrHandlerInterrupt is returned when a handler that was given returns and error
+var ErrHandlerInterrupt = errors.New("Handler returned an error and interrupted iteration")
```

Comment:
This an an error that **must** be handled... It's non-fatal... So the caller has the context..

They are easier to handle when done like this... Hence, why I'm willing to not include any context